src/Extractor.ts

```diff
@@ -426,7 +465,16 @@ class Extractor {
     const name = this.entityName(node, tag);
     if (name == null) return null;
 
-    if (!ts.isUnionTypeNode(node.type)) {
+    if (ts.isTypeReferenceNode(node)) {
+      if (this.hasTag(node, EXTERNAL_TAG)) {
+        const externalImportPath = this.externalModule(node);
+        if (externalImportPath) {
+          this.recordTypeName(node, name, "UNION", externalImportPath);
+          return;
+        }
```

Comment:
I think `externalImportPath` is only null in the error case, so we should bail in that case.
```suggestion
        const externalImportPath = this.externalModule(node);
        if (externalImportPath == null) return;
        this.recordTypeName(node, name, "UNION", externalImportPath);
        return;
```

This pattern appears in a few places.

Another option might be to pass more context to `this.externalModule` and move the `this.recordTypeName` into `this.externalModule`. The, if you have an `EXTERNAL_TAG` you could just do: `return this.externalModule(node, name "UNION");`